engine/engine/src/engine.cpp

```diff
@@ -852,48 +844,24 @@ namespace dmEngine
         engine->m_InvPhysicalWidth = 1.0f / physical_width;
         engine->m_InvPhysicalHeight = 1.0f / physical_height;
 
-        engine->m_UseSwVsync = false;
-
 #if defined(__MACH__) || defined(__linux__) || defined(_WIN32)
         engine->m_RunWhileIconified = dmConfigFile::GetInt(engine->m_Config, "engine.run_while_iconified", 0);
 #endif
 
         dmGameSystem::OnWindowCreated(physical_width, physical_height);
 
-        bool setting_vsync = dmConfigFile::GetInt(engine->m_Config, "display.vsync", true);
-        uint32_t setting_update_frequency = dmConfigFile::GetInt(engine->m_Config, "display.update_frequency", 0);
-        uint32_t update_frequency = setting_update_frequency;
-        uint32_t swap_interval = 1;
+        engine->m_UpdateFrequency = dmConfigFile::GetInt(engine->m_Config, "display.update_frequency", 0);
+
+        SetUpdateFrequency(engine, engine->m_UpdateFrequency);
+
+        bool setting_vsync = dmConfigFile::GetInt(engine->m_Config, "display.vsync", true); // Deprecated
 
+        uint32_t opengl_swap_interval = dmConfigFile::GetInt(engine->m_Config, "display.swap_interval", 1);
```

Comment:
Is this only for opengl? What about vulkan?

Reply:
You cannot set swap interval on Vulkan.
There are different present modes for vulkan, and we currently use the FIFO mode. We might want to expose that as a setting later on (when someone requests it)

Reply:
I feel there's a danger in coupling the two backend behaviors too closely.

Reply:
Ok, got it. Makes sense!